veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,290 @@
+
+> [!NOTE] 
+> In the future, it may be desirable to implement this simple logic in Go rather than call an external binary. 
+> However, for the initial introduction of this functionality it will be simpler and safer to have both client and severer released and tested together under the same package.    
+
+
+#### Integration with virt-handler
+
+To support this process, virt-handler will be modified to invoke `passt-repair` in a dedicated goroutine. This goroutine will
+run to completion or timeout, ensuring the migration process does not hang indefinitely.
+
+The invocation is conditioned to the VMI having an interface with `passt` network binding and an enabled feature gate:
+
+1. Before migration starts (source node) in [vmUpdateHelperMigrationSource](https://github.com/kubevirt/kubevirt/blob/release-1.5/pkg/virt-handler/vm.go#L2714) to set the TCP_REPAIR socket option.
+2. Before migration finalizes (target node) in [finalizeMigration](https://github.com/kubevirt/kubevirt/blob/release-1.5/pkg/virt-handler/vm.go#L2714) to set and then clear the TCP_REPAIR socket option.
+
+##### Failure Scenarios:
+
+passt-repair can fail or timeout if:
+- The Unix Domain Socket created by passt isn't found.
+- Communication with passt is delayed or unresponsive.
+
+Either scenario likely results in broken TCP connections for the guest VM, requiring reconnection.
+- virt-handler does not fail migration on passt-repair errors.
+- passt currently does not abort migration upon TCP_REPAIR failure.
+
+> [!NOTE]
+> Success of passt-repair doesn't guarantee seamless migration alone; sticky IPs from the network provider and CNI are essential.
+
+### passt network binding plugin 
+
+The binding of passt is implemented as a [network binding plugin](https://kubevirt.io/user-guide/network/network_binding_plugins/#network-binding-plugins).
+It comprises:
+- A [sidecar container](#Sidecar container) running in the virt-launcher pod. Its sole responsibility is to mutate the DomainXML.
+- A [CNI Plugin](#CNI plugin) that is invoked during pod setup/teardown. 
+- Configuration.
+Further details can be found in the [design document](https://github.com/kubevirt/community/blob/main/design-proposals/network-binding-plugin/network-binding-plugin.md#design-details).
+
+#### Sidecar container
+
+`vhost-user` enables high-performance communication between virtual machines and user-space processes, typically offloading VirtIO emulation to dedicated user-space processes (e.g., DPDK or passt). 
+It improves I/O performance through zero-copy mechanisms and reduced context-switching overhead. Additionally, it is required for seamless TCP migration.
+
+`vhost-user` must be enabled in the DomainXML. Small code changes are required in passt sidecar [code](https://github.com/kubevirt/kubevirt/tree/release-1.5/cmd/sidecars/network-passt-binding). The format is described [here](https://libvirt.org/formatdomain.html#vhost-user-connection-with-passt-backend).
```

Comment:
Are the required changes already implemented by the libvirt version used by KubeVirt?

Reply:
Yes, see https://github.com/kubevirt/kubevirt/pull/14101.